### ISSUE TITLE
Only the objects in current list page is shown.

### ISSUE THREAD

**Issue by byenilmez** [dropseed/django-admincharts#25]
If I have a model with 150 objects, only first 100 of them are included in chart. And I browse to next page, following 50 records are included. Is this by design?

If not, using `.queryset` instead of `.result_list` on line 51 of admincharts/admin.py fixes the issue for me.

**Comment by davegaeddert** [comment]
Hey @byenilmez — I haven't looked at this recently but I do think it's intentional. See the last paragraph on the README:
> By default, the objects in your chart will be the objects that are currently visible in your list view.
This means that admin controls like [search](https://docs.djangoproject.com/en/3.2/ref/contrib/admin/#django.contrib.admin.ModelAdmin.search_fields) and [list filter](https://docs.djangoproject.com/en/3.2/ref/contrib/admin/#django.contrib.admin.ModelAdmin.list_filter) will update your chart,
and you can use the Django [pagination](https://docs.djangoproject.com/en/3.2/ref/contrib/admin/#django.contrib.admin.ModelAdmin.list_per_page) [settings](https://docs.djangoproject.com/en/3.2/ref/contrib/admin/#django.contrib.admin.ModelAdmin.list_max_show_all) to control how many objects you want in your chart at a time.
If you want, you can also sidestep the list queryset entirely by using overriding `get_list_chart_queryset`.

You can double-check this, but I think if you use `.queryset` it won't respect the search or list filters. I think by default, the chart is more useful when it does (I use list filters all the time, so it's essentially a way to update the chart data and show something more specific).

Does that help? You should be able to override `get_list_chart_queryset` to return a different set of objects if you want, but looking at it now I'm wondering if it needs another argument or two to make it easier to use...

**Comment by byenilmez** [comment]
Hi Dave. I totally missed the pagination part in the readme, sorry about that.

For my use case, I need the entire queryset (I show the total number of orders for each user in orders model). I just checked again, `.queryset` indeed returns filtered queryset before pagination.

Overriding `get_list_chart_queryset `does not work in my case as I want to keep the filters.

I agree additional arguments will make it more flexible. Alternatively, `get_list_chart_queryset` can be called using `changelist` instead of `changelist.result_list` and default `get_list_chart_queryset` function can return `changelist.result_list`. So, overriding this function can allow use of entire queryset.

i.e.,
```
...
    def changelist_view(self, request, extra_context=None):
        ...
            chart_queryset = self.get_list_chart_queryset(changelist)
...
```

```
    def get_list_chart_queryset(self, change_list):
        """Returns the current changelist results by default"""
        return change_list.result_list
```

Then overriding function as below this will give entire queryset with filters applied:
```
    def get_list_chart_queryset(self, change_list):
        """Returns the current changelist results by default"""
        return change_list.queryset
```

**Comment by davegaeddert** [comment]
I had the same thought on just passing the changelist itself. So #26 will give you what you need, right?

Happy to merge and release it if that's the case!

**Comment by byenilmez** [comment]
Yes, this is perfect! Thanks a lot!

**Comment by davegaeddert** [comment]
Released as 0.3.0!
